src/main/java/com/bravo/user/dao/model/mapper/ResourceMapper.java

```diff
@@ -41,17 +39,6 @@ public AddressDto convertAddress(final Address address){
     return dto;
   }
 
-  public <T extends Collection<Payment>> List<PaymentDto> convertPayments(final T payments){
-    return payments.stream().map(this::convertPayment).collect(Collectors.toList());
-  }
-
-  public PaymentDto convertPayment(final Payment payment){
```

Comment:
Can you elaborate on why the new `PaymentMapper` class is superior to these methods within the existing `ResourceMapper` class?

Reply:
@wheelerswebservices because mapperFacade, which is used here is not the flexible one from SOLID principles perpective.
First of all, if you would have 10 dto's which you need to map from the entity its not a good idea to create 20 methods inside ResourseMapper class.
Second, is that you need to implement all of the extra mapping, or when field names are different. In the Mapstruct its simple, you only need an interface, and specify everything only using annotations. Its very useful and simple to understand. Its flexible when you need some extra mappings, for example as was for card mask. So everything for this mapping is located only in 1 Java interface.
Thats why mapstruct is so popular today.
You could read all the details below: https://www.baeldung.com/java-performance-mapping-frameworks